### PR TITLE
Text CNN

### PR DESCRIPTION
**Pull request by miaecle**
This PR adds in:

- Text CNN on smiles strings: reimplementation of the discriminator module in ORGAN ([arXiv:1705.10843](https://arxiv.org/abs/1705.10843))
, derived from Convolutional Neural Networks for Sentence Classification (Yoon Kim, http://emnlp2014.org/papers/pdf/EMNLP2014181.pdf)

### CODE REVIEW

**Comment by peastman** [comment]
A lot more documentation would be really helpful.  What do these classes do?  What are they used for?

File: deepchem/models/tensorgraph/layers.py [hunk 1]
```diff
@@ -480,6 +480,55 @@ def _get_scope_name(self):
       return self._shared_with._get_scope_name()
 
 
+class Highway(Layer):
```
Comment (rbharath) [inline deepchem/models/tensorgraph/layers.py]: Could you add a description of Highway layers and why they're useful here?

File: deepchem/models/tensorgraph/layers.py [hunk 2]
```diff
@@ -1516,6 +1565,35 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
     return out_tensor
 
 
+class MaxPool1D(Layer):
+
+  def __init__(self, window_shape=2, strides=1, padding="SAME", **kwargs):
```
Comment (rbharath) [inline deepchem/models/tensorgraph/layers.py]: Could you add a docstring here?

File: deepchem/models/tensorgraph/layers.py [hunk 3]
```diff
@@ -1516,6 +1565,35 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
     return out_tensor
 
 
+class MaxPool1D(Layer):
```
Comment (rbharath) [inline deepchem/models/tensorgraph/layers.py]: Could you add a high level description here?

File: deepchem/models/tensorgraph/models/text_cnn.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python2
```
Comment (rbharath) [inline deepchem/models/tensorgraph/models/text_cnn.py]: Maybe remove this line? This file probably shouldn't be command line executable.

File: deepchem/models/tensorgraph/models/text_cnn.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    'c': 28,
+    'Cl': 29,
+    'Br': 30,
+    'n': 31,
+    'o': 32,
+    's': 33
+}
+
+
+class TextCNNTensorGraph(TensorGraph):
```
Comment (rbharath) [inline deepchem/models/tensorgraph/models/text_cnn.py]: Could you add a docstring providing a high level description of the purpose of this class?

**Comment by miaecle** [comment]
@peastman @rbharath Thanks for the review, have more docstrings added!

File: deepchem/models/tensorgraph/layers.py (hunk 1 above)
Comment (miaecle) [reply deepchem/models/tensorgraph/layers.py]: Just added. I feel that Highway layer is like a simplified version of GRU, I implemented it here to repeat the work in the ORGAN paper, though I have not done a detailed comparison.

File: deepchem/models/tensorgraph/layers.py (hunk 3 above)
Comment (miaecle) [reply deepchem/models/tensorgraph/layers.py]: Just added!

File: deepchem/models/tensorgraph/models/text_cnn.py (hunk 5 above)
Comment (miaecle) [reply deepchem/models/tensorgraph/models/text_cnn.py]: Added a description of model and references.

**Comment by peastman** [comment]
I think the documentation could still be made a lot more useful.  It's not just a matter of giving equations for what the class does.  It's about explaining what the class is useful for, when you would want to use it, and how to use it correctly.  Take a look at https://github.com/deepchem/deepchem/blob/master/deepchem/models/tensorgraph/models/seqtoseq.py for an example of the sort of documentation I mean.

**Comment by miaecle** [comment]
@peastman Sorry for the delay, just get some more docs added. This is a pretty simple model so I just added some instructions and comments on how to use the model.

**Comment by rbharath** [comment]
I think this is failing due to #877

**Comment by miaecle** [comment]
@rbharath Looks like everything is all set.

**Comment by rbharath** [comment]
LGTM

**Pull request merged**